File: app/src/main/java/com/iita/akilimo/utils/enums/EnumOperationType.java
package com.iita.akilimo.utils.enums;

public enum EnumOperationType {
    MANUAL {
        @Override
        public String operationName() {
            return "manual";
        }
    },
    MECHANICAL {
        @Override
        public String operationName() {
            return "tractor";
        }
    },
    NONE {
        @Override
        public String operationName() {
            return "NA";
        }
    };


    public abstract String operationName();
}

File: app/src/main/java/com/iita/akilimo/utils/enums/EnumOperationType.kt
package com.iita.akilimo.utils.enums

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
enum class EnumOperationType : Parcelable {
    MANUAL {
        override fun operationName(): String {
            return "manual"
        }
    },
    MECHANICAL {
        override fun operationName(): String {
            return "tractor"
        }
    },
    HERBICIDE {
        override fun operationName(): String {
            return "herbicide"
        }
    },
    NONE {
        override fun operationName(): String {
            return "NA"
        }
    };

    abstract fun operationName(): String
}
